controllers/operatorconfigmap_controller.go

```diff
@@ -0,0 +1,484 @@
+type OperatorConfigMapReconciler struct {
+	client.Client
+	OperatorDeployment *appsv1.Deployment
+	OperatorNamespace  string
+	ConsolePort        int32
+	Scheme             *runtime.Scheme
+
+	log               logr.Logger
+	ctx               context.Context
+	consoleDeployment *appsv1.Deployment
+	cephFSDeployment  *appsv1.Deployment
+	cephFSDaemonSet   *appsv1.DaemonSet
+	rbdDeployment     *appsv1.Deployment
+	rbdDaemonSet      *appsv1.DaemonSet
+	scc               *secv1.SecurityContextConstraints
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (c *OperatorConfigMapReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	clusterVersionPredicates := builder.WithPredicates(
+		predicate.GenerationChangedPredicate{},
+	)
+
+	configMapPredicates := builder.WithPredicates(
+		predicate.NewPredicateFuncs(
+			func(client client.Object) bool {
+				namespace := client.GetNamespace()
+				name := client.GetName()
+				return ((namespace == c.OperatorNamespace) && (name == operatorConfigMapName))
+			},
+		),
+	)
+	// Reconcile the ClusterVersion object when the operator config map is updated
+	enqueueConfigMapRequest := handler.EnqueueRequestsFromMapFunc(
+		func(_ context.Context, client client.Object) []reconcile.Request {
+			return []reconcile.Request{{
+				NamespacedName: types.NamespacedName{
+					Name:      operatorConfigMapName,
+					Namespace: c.OperatorNamespace,
+				},
+			}}
+		},
+	)
+
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&corev1.ConfigMap{}, configMapPredicates).
+		Watches(&configv1.ClusterVersion{}, enqueueConfigMapRequest, clusterVersionPredicates).
+		Complete(c)
+}
+
+//+kubebuilder:rbac:groups=config.openshift.io,resources=clusterversions,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=config.openshift.io,resources=clusterversions/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=config.openshift.io,resources=clusterversions/finalizers,verbs=update
+//+kubebuilder:rbac:groups="apps",resources=deployments,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups="apps",resources=deployments/finalizers,verbs=update
+//+kubebuilder:rbac:groups="apps",resources=daemonsets,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups="apps",resources=daemonsets/finalizers,verbs=update
+//+kubebuilder:rbac:groups="storage.k8s.io",resources=csidrivers,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups="",resources=configmaps,verbs=get;list;watch;create;update;delete
+//+kubebuilder:rbac:groups=security.openshift.io,resources=securitycontextconstraints,verbs=get;list;watch;create;patch;update
+//+kubebuilder:rbac:groups=monitoring.coreos.com,resources=prometheusrules,verbs=get;list;watch;create;update
+//+kubebuilder:rbac:groups="",resources=services,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=console.openshift.io,resources=consoleplugins,verbs=*
```

Comment:
just realized, many are high privileged rbac, pls raise an issue to revisit this at a later point